lib/fluent/plugin/out_mongo.rb

```diff
@@ -227,35 +184,26 @@ def format_collection_name(collection_name)
       formatted
     end
 
-    def get_or_create_collection(collection_name)
-      collection_name = format_collection_name(collection_name)
-      return @clients[collection_name] if @clients[collection_name]
-
-      @db ||= get_connection
-      if @db.collection_names.include?(collection_name)
-        collection = @db.collection(collection_name)
-        unless @disable_collection_check
-          capped = collection.capped?
-          unless @collection_options[:capped] == capped # TODO: Verify capped configuration
-            new_mode = format_collection_mode(@collection_options[:capped])
-            old_mode = format_collection_mode(capped)
-            raise ConfigError, "New configuration is different from existing collection: new = #{new_mode}, old = #{old_mode}"
+    def operate(collection, records)
+      begin
+        if @replace_dot_in_key_with
+          records.map! do |r|
+            replace_key_of_hash(r, ".", @replace_dot_in_key_with)
+          end
+        end
+        if @replace_dollar_in_key_with
+          records.map! do |r|
+            replace_key_of_hash(r, /^\$/, @replace_dollar_in_key_with)
           end
         end
-      else
-        collection = @db.create_collection(collection_name, @collection_options)
-      end
-
-      @clients[collection_name] = collection
-    end
-
-    def format_collection_mode(mode)
-      mode ? 'capped' : 'normal'
-    end
 
-    def get_connection
-      db = Mongo::MongoClient.new(@host, @port, @connection_options).db(@database)
-      authenticate(db)
+        @client[collection, @collection_options].insert_many(records)
```

Comment:
New MongoDB Ruby driver does not have such a concrete error fields like *writeErrors*.
ref: http://api.mongodb.com/ruby/2.2.3/Mongo/BulkWrite/Result.html

Example `Result` output is like this:

```log
#<Mongo::BulkWrite::Result:0x007fcb31fc4578 @results={"n_inserted"=>2, "n"=>0, "inserted_ids"=>[BSON::ObjectId('57bfc69af4a32a4b233fa2b7'), BSON::ObjectId('57bfc69af4a32a4b233fa2b8')]}>
```

And I proposed to change this behaviour, but it was rejected due to this MongoDB specification:
https://github.com/mongodb/specifications/blob/04213be3f7ba9b9cb2943237435e63418ab82683/source/crud/crud.rst#results

Any thoughts?